ambry-commons/src/main/java/com/github/ambry/commons/HardwareUsageMeter.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.commons;
+
+import java.lang.management.BufferPoolMXBean;
+import java.lang.management.ManagementFactory;
+import com.sun.management.OperatingSystemMXBean;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A class to report hardware usage.
+ */
+public class HardwareUsageMeter {
+  private static final Logger logger = LoggerFactory.getLogger(HardwareUsageMeter.class);
+  private long cpuRecordTime = 0;
+  private final int cpuMaxAllowedDelayInMs;
+  private int cpuPercentage;
+  private long memoryRecordTime = 0;
+  private final int memoryMaxAllowedDelayInMs;
+  private int memoryPercentage;
+  private long maxMemory = Runtime.getRuntime().maxMemory();
+  private BufferPoolMXBean directMemoryMxBean = null;
```

Comment:
I guess it will be null even if you don't assign.

Reply:
fixed.